src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -768,11 +768,13 @@ private void DynamoViewModelChangeScaleFactor(object sender, EventArgs e)
             var view = new Prompts.ChangeScaleFactorPrompt(dynamoViewModel.ScaleFactorLog) { Owner = this };
             if (view.ShowDialog() == true)
             {
-                if (dynamoViewModel.ScaleFactorLog != view.SliderValue)
+                if (dynamoViewModel.ScaleFactorLog != view.ScaleValue)
                 {
-                    dynamoViewModel.ScaleFactorLog = view.SliderValue;
+                    dynamoViewModel.ScaleFactorLog = view.ScaleValue;
                     dynamoViewModel.CurrentSpace.HasUnsavedChanges = true;
 
+                    Log("Scale factor log changed to " + view.ScaleValue);
```

Comment:
I'm thinking if you could have something like this in `ChangeScaleFactorPrompt` class:
```
enum Size
        {
            Medium, 
            Small,
            Large,
            ExtraLarge
        }

        public Tuple<double, double> ScaleRange { get; }
        private Dictionary<Size, Tuple<double, double>> scaleRanges = new Dictionary<Size, Tuple<double, double>>
        {
            {Size.Small, new Tuple<double, double>(0.000001, 100)},
            {Size.Medium, new Tuple<double, double>(0.0001, 10000)},
            {Size.Large, new Tuple<double, double>(0.01, 1000000)},
            {Size.ExtraLarge, new Tuple<double, double>(0.01, 1000000)}
        };
```
... and then set `ScaleRange` to `ScaleRanges[Size.XYZ]` whenever the respective `XYZ` size button is clicked, then call:
```
Log(String.Format("Geometry working range changed to ({0}, {1})", view.ScaleRange.Item2, view.ScaleRange.Item2));
```